baybe/recommenders/bayesian.py

```diff
@@ -417,40 +415,28 @@ def _recommend_hybrid(
                 f"acquisition functions."
             ) from ex
 
-        # >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
```

Comment:
I might overlook something here but I think you missed the point of the markers, there was a good reason why I added them: the entire content can be removed after the fix. If I'm not mistaken, you should be able to use directly the output from the botorch routine (and wrap it appropriately as a DataFrame). Your current logic is to extract the subparts to only reassemble the joint part again. This was necessary only for the workaround because the discrete part needed special treatment. But please correct me if there is something I don't see (haven't checked in details, this is only what I remember from when I added the workaround)

Reply:
hybrid_recommend is expected to return the candidates in exp representation
for that I need to do the join since botorch oeprates and returns candidates in comp rep, and 
the join should/can/only makes sense only happen on the discrete parts
hence I still need to separate the disc/conti parts

im not reassembling the subparts in comp rep (which would be futile as you say) but in exp rep

Reply:
another way to see it: ive literally just change the way we get the indices (now=merge, before=manhattan distance), theres no need to change anything else, so to me it seems the markers were wider than necessary

Reply:
I guess you are right. Have had a look and the code and agree 👍🏼 Good that we talked about it 😄